### ISSUE TITLE
Determining the type of a (parsed) transaction

### ISSUE THREAD

**Issue by eduarrrd** [WoLpH/mt940#70]
I might be missing something obvious here but once a `models.Transaction` object is created, how do I know what type it is? I would have guessed there would some sort of reference to a `tags.*` class/object/singleton in the `Transaction.data` dict but I'm not seeing it. Maybe I'm simply misunderstanding the architecture here ...

My use case is that I have full dump of an account but I only care about certain entries (specifically `Statement`s). I understand I could check for the presence of certain keys (e.g. `guessed_entry_date`) but that seems backwards, error prone, and fragile to me.

**Comment by WoLpH** [comment]
I don't really get what you mean by type? Do you mean the distiction between a debit and credit transaction?
Or are you referring to missing fields if the file has incomplete transactions?

If the transactions are indeed incomplete than the data will be missing from the transaction. The library doesn't guess what should be there but simply returns whichever data it received.

Any approach is unfortunately error-prone due to the many inconsistencies exist between the implementations from different banks. Checking for properties that you expect to be there is probably the safest method and the least fragile option in any case. If you have a different suggestion though, I'm open to improvements :)

**Comment by eduarrrd** [comment]
Sorry, for the confusion, please let me rephrase:

Once I have a `Transaction` object, can I safely assume that it can only represent a "Statement Line" (i.e. tag :61:) from the original mt940 file and nothing else?

The reason I'm asking is b/c some of the transactions I looked at did not have `guessed_entry_date` populated despite check [1] having to succeed as both fields are mandatory according to the specs I've seen.

In any case I'll dig deeper later today.

[1] https://github.com/WoLpH/mt940/blob/c69ba92272506def7ef33150c576a522943f10a5/mt940/tags.py#L308

**Comment by WoLpH** [comment]
That's pretty much it:

https://github.com/WoLpH/mt940/blob/1188a54171c64469b9845b518a5f233dc0500a53/mt940/models.py#L428-L456
